core/src/main/java/org/apache/accumulo/core/conf/AccumuloConfiguration.java

```diff
@@ -597,4 +599,13 @@ public T derive() {
    * this configuration.
    */
   public void invalidateCache() {}
+
+  /**
+   * get a parent configuration or null if it does not exist.
+   *
+   * @since 2.1.0
+   */
+  public AccumuloConfiguration getParent() {
+    return null;
```

Comment:
Should this return something?

Reply:
Sub-classes that extend AccumuloConfiguration that have a hierarchy,  override this method to return a parent - for  ones that are stand alone (like DefaultConfiguration and ConfigurationCopy) then there is no hierarchy and null is appropriate.  The alternative would be to declare the method abstract and force each sub-class to provide an implementation. So the null is appropriate, it just depends on where the responsibility for returning null is implemented. Here, or the sub-classes.

Reply:
Ok. I think this should probably throw an exception then. If the class doesn't have a parent, do we want `getParent()` to return null?

Reply:
When would it be appropriate to throw an Exception and what exception type do you suggest?

The code that uses Accumulo configuration blindly calls getParent to walk up the hierarchy if it exists. For the existing cases that do not have a parent (DefaultConfiguration, ConfigurationCopy) there is no parent. In cases like system, namespace and table, calling getParent returns the next configuration up the chain.  Getting a null indicates that the top of the hierarchy has been reached and is not an exception. When would an exception be appropriate?